lib/elixir/src/elixir_tokenizer.erl

```diff
@@ -165,51 +168,55 @@ tokenize([$~,S,H|T] = Original, Line, Scope, Tokens) when ?is_sigil(H), ?is_upca
+tokenize([$?,$\\,$x,${,A,B,C,D,E,F,$}|T], Line, Column, Scope, Tokens)
     when ?is_hex(A), ?is_hex(B), ?is_hex(C), ?is_hex(D), ?is_hex(E), ?is_hex(F) ->
   Char = escape_char([$\\,$x,${,A,B,C,D,E,F,$}]),
-  tokenize(T, Line, Scope, [{number, Line, Char}|Tokens]);
+  tokenize(T, Line, Column + 12, Scope, [{number, {Line, Column, Column + 12}, Char}|Tokens]);
 
-tokenize([$?,$\\,$x,${,A,B,C,D,E,$}|T], Line, Scope, Tokens)
+tokenize([$?,$\\,$x,${,A,B,C,D,E,$}|T], Line, Column, Scope, Tokens)
     when ?is_hex(A), ?is_hex(B), ?is_hex(C), ?is_hex(D), ?is_hex(E) ->
   Char = escape_char([$\\,$x,${,A,B,C,D,E,$}]),
-  tokenize(T, Line, Scope, [{number, Line, Char}|Tokens]);
+  tokenize(T, Line, Column + 11, Scope, [{number, {Line, Column, Column + 11}, Char}|Tokens]);
 
-tokenize([$?,$\\,$x,${,A,B,C,D,$}|T], Line, Scope, Tokens)
+tokenize([$?,$\\,$x,${,A,B,C,D,$}|T], Line, Column, Scope, Tokens)
     when ?is_hex(A), ?is_hex(B), ?is_hex(C), ?is_hex(D) ->
   Char = escape_char([$\\,$x,${,A,B,C,D,$}]),
-  tokenize(T, Line, Scope, [{number, Line, Char}|Tokens]);
+  tokenize(T, Line, Column + 10, Scope, [{number, {Line, Column, Column + 10}, Char}|Tokens]);
 
-tokenize([$?,$\\,$x,${,A,B,C,$}|T], Line, Scope, Tokens)
+tokenize([$?,$\\,$x,${,A,B,C,$}|T], Line, Column, Scope, Tokens)
     when ?is_hex(A), ?is_hex(B), ?is_hex(C) ->
   Char = escape_char([$\\,$x,${,A,B,C,$}]),
-  tokenize(T, Line, Scope, [{number, Line, Char}|Tokens]);
+  tokenize(T, Line, Column + 9, Scope, [{number, {Line, Column, Column + 9}, Char}|Tokens]);
 
-tokenize([$?,$\\,$x,${,A,B,$}|T], Line, Scope, Tokens)
+tokenize([$?,$\\,$x,${,A,B,$}|T], Line, Column, Scope, Tokens)
     when ?is_hex(A), ?is_hex(B) ->
   Char = escape_char([$\\,$x,${,A,B,$}]),
-  tokenize(T, Line, Scope, [{number, Line, Char}|Tokens]);
+  tokenize(T, Line, Column + 8, Scope, [{number, {Line, Column, Column + 8}, Char}|Tokens]);
 
-tokenize([$?,$\\,$x,${,A,$}|T], Line, Scope, Tokens)
+tokenize([$?,$\\,$x,${,A,$}|T], Line, Column, Scope, Tokens)
     when ?is_hex(A) ->
   Char = escape_char([$\\,$x,${,A,$}]),
-  tokenize(T, Line, Scope, [{number, Line, Char}|Tokens]);
+  tokenize(T, Line, Column + 7, Scope, [{number, {Line, Column, Column + 7}, Char}|Tokens]);
 
-tokenize([$?,$\\,$x,A,B|T], Line, Scope, Tokens)
+tokenize([$?,$\\,$x,A,B|T], Line, Column, Scope, Tokens)
     when ?is_hex(A), ?is_hex(B) ->
   Char = escape_char([$\\,$x,A,B]),
-  tokenize(T, Line, Scope, [{number, Line, Char}|Tokens]);
+  tokenize(T, Line, Column + 6, Scope, [{number, {Line, Column, Column + 6}, Char}|Tokens]);
 
-tokenize([$?,$\\,$x,A|T], Line, Scope, Tokens)
+tokenize([$?,$\\,$x,A|T], Line, Column, Scope, Tokens)
     when ?is_hex(A) ->
   Char = escape_char([$\\,$x,A]),
-  tokenize(T, Line, Scope, [{number, Line, Char}|Tokens]);
+  tokenize(T, Line, Column + 5, Scope, [{number, {Line, Column, Column + 5}, Char}|Tokens]);
 
-tokenize([$?,$\\,H|T], Line, Scope, Tokens) ->
+tokenize([$?, $\\, $\\|T], Line, Column, Scope, Tokens) ->
```

Comment:
This one is just 3 characters... we use `$\\` to escape the erlang character but in the code it is only a single `\`. We could rather get rid of this clause.